entity-framework/core/testing/choosing-a-testing-strategy.md

```diff
@@ -0,0 +1,112 @@
+
+One possible disadvantage of the repository pattern is that it requires you to architect your application accordingly: application code can no longer execute LINQ queries directly, but must instead call into a repository method, which executes the query (and is stubbed in testing). This can make for a somewhat heavier application architecture, but can also have other advantages: all data access code is concentrated in one place rather than being spread around the application, and if your application needs to support more than one database, the repository abstraction can be very helpful for tweaking queries across providers.
+
+For an example of a unit testing with a repository, see the [unit testing page](xref:core/testing/unit-testing#repository-pattern).
+
+## Overall comparison
+
+The following table provides a quick, comparative view of the different testing techniques, and shows which functionality can be tested under which approach:
+
+Feature                             | InMemory     | SQLite in-memory          | Mock DbContext | Repository pattern | Integration testing
+----------------------------------- | ------------ | ------------------------- | -------------- | ------------------ | -------------------
+Test double type                    | Fake         | Fake                      | Mock/stub      | Mock/stub          | Real, no double
+Raw SQL?                            | No           | Depends                   | No             | Yes                | Yes
+Transactions?                       | No (ignored) | Yes                       | Yes            | Yes                | Yes
+Provider-specific translations?     | No           | No                        | No             | Yes                | Yes
+Exact query behavior?               | Depends      | Depends                   | Depends        | Yes                | Yes
+Requires special code architecture? | No           | No                        | No             | Yes                | No
+
+## Summary
+
+* We highly recommend that developers have good test coverage of their application running against their actual production database system. This provides confidence that the application actually works in production, and with proper design, tests can execute reliably and quickly. Since integration tests are required in any case, it's a good idea to start there, and add unit tests later, based on need.
+* For unit testing, we recommend implementing the repository pattern, allowing you to properly stub or mock out your data access layer above EF Core, rather than via a fake EF Core provider (Sqlite/InMemory) or by mocking `DbSet`.
```

Comment:
```suggestion
* If unit tests for queries are absolutely required, then we recommend implementing the repository pattern, allowing you to properly stub or mock out your data access layer above EF Core, rather than via a fake EF Core provider (Sqlite/InMemory) or by mocking `DbSet`.
```

Reply:
That's really strong - I tried to not be that categorical about rejecting unit testing. Can we compromise and remove the "absolutely"? :rofl: